test/src/e2e_vm_tests/test_programs/should_pass/language/nested_struct_destructuring/src/main.sw

```diff
@@ -0,0 +1,21 @@
+script;
+
+// Tests nested struct destructuring
+
+fn main() -> u64 {
+    let point1 = Point { x: 0, y: 0 };
+    let point2 = Point { x: 1, y: 1 };
+    let line = Line { p1: point1, p2: point2 };
+    let Line { p1: Point { x: x0, y: y0 }, p2: Point { x: x1, y: y1} } = line;
```

Comment:
alright, don't hate me, but what about a nested tuple in a struct and vice versa?

Reply:
Yeah I can add a test for that.  Should I also include an example for the docs?

Reply:
Docs example could be cool, to show people just how deep destructuring can go!